fili-core/src/main/java/com/yahoo/bard/webservice/table/BaseSchema.java

```diff
@@ -33,6 +33,22 @@ protected BaseSchema(Granularity granularity, Iterable<Column> columns) {
         return columns;
     }
 
+    /**
+     * Retrieve a column from schema given the name of the column.
+     *
+     * @param columnName  The column name in string
+     *
+     * @return the column if found
+     */
+    public Column getColumn(String columnName) {
```

Comment:
Since this is a brand new method, we should have it return `Optional<Column>` rather than potentially `null`. Optionals are nice because they bake into the type system the fact that there may not be a column, and so it forces the callers to deal with that eventuality.

Also, you can stream this very easily:

```public Optional<Column> getColumn(String columnName) {
    return columns.stream()
              .filter(column -> column.getName().equals(columnName))
              .findFirst();
}
```
The beautiful thing is that `findFirst` returns an `Optional`, so you can kill two birds with one stone!